### PR TITLE
fix: ignore content-length when dumping HEAD

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: lib/api/api-request.js [hunk 1]
```diff
@@ -111,7 +112,13 @@ class RequestHandler extends AsyncResource {
...
     const contentType = parsedHeaders['content-type']
     const contentLength = parsedHeaders['content-length']
-    const body = new Readable({ resume, abort, contentType, contentLength, highWaterMark })
+    const body = new Readable({
+      resume,
+      abort,
+      contentType,
+      contentLength: this.method === 'HEAD' ? null : contentLength,
+      highWaterMark
+    })
```
Comment (Uzlopak) [inline lib/api/api-request.js:121]: why not the following?
```js
    const contentLength = method === 'HEAD' ? null : parsedHeaders['content-length'] 
    const body = new Readable({
      resume,
      abort,
      contentType,
      contentLength,
      highWaterMark
    })
```

File: lib/api/api-request.js [hunk 2]
```diff
@@ -111,12 +111,13 @@ class RequestHandler extends AsyncResource {
...
     const parsedHeaders = responseHeaders === 'raw' ? util.parseHeaders(rawHeaders) : headers
     const contentType = parsedHeaders['content-type']
-    const contentLength = parsedHeaders['content-length']
     const body = new Readable({
       resume,
       abort,
       contentType,
-      contentLength: this.method === 'HEAD' ? null : contentLength,
+      contentLength: this.method !== 'HEAD' && parsedHeaders['content-length']
+        ? Number(parsedHeaders['content-length'])
```
Comment (Uzlopak) [inline lib/api/api-request.js:119]: is it ensured, that it is a positive integer?

Comment (ronag) [reply lib/api/api-request.js:119]: No, but does it matter?

**Review by mcollina — CHANGES_REQUESTED** [review]
+1 to the change. Can you please add a unit test?

File: lib/api/api-request.js (hunk 2 above)
Comment (Uzlopak) [reply lib/api/api-request.js:119]: what happens if it is NaN?

Comment (metcoder95) [reply lib/api/api-request.js:119]: If the server is fully compliant, it should never sent any non-positive value; if a non-positive value is sent, the parser will fail with a non-compliant error

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**